src/openrct2/actions/FootpathPlaceAction.hpp

```diff
@@ -182,15 +183,26 @@ DEFINE_GAME_ACTION(FootpathPlaceAction, GAME_COMMAND_PLACE_PATH, GameActionResul
         }
 
         pathElement->SetSurfaceEntryIndex(_type & ~FOOTPATH_ELEMENT_INSERT_QUEUE);
-        if (_type & FOOTPATH_ELEMENT_INSERT_QUEUE)
-        {
-            pathElement->SetIsQueue(true);
-        }
-        else
+        bool isQueue = _type & FOOTPATH_ELEMENT_INSERT_QUEUE;
+        pathElement->SetIsQueue(isQueue);
+
+        rct_scenery_entry* elem = pathElement->GetAdditionEntry();
+        if (elem != nullptr)
         {
-            pathElement->SetIsQueue(false);
+            if (isQueue)
+            {
+                //remove any addition that isn't a TV
+                if ((elem->path_bit.flags & PATH_BIT_FLAG_IS_QUEUE_SCREEN) == 0)
+                    pathElement->SetAddition(0);
+            }
+            else
+            {
+                //remove all TVs
+                if ((elem->path_bit.flags & PATH_BIT_FLAG_IS_QUEUE_SCREEN) != 0)
+                    pathElement->SetAddition(0);
+            }
         }
-        pathElement->SetAddition(0);
+        
         pathElement->SetIsBroken(false);
```

Comment:
Maybe this line should be removed as well? (Or maybe moved inside the conditional between lines 196 and 197, to ensure that the broken flag gets set false when non-TV path items are removed. Since TV's can't be broken -- and therefore shouldn't have that flag in the first place -- we don't need to worry about that part.)

If changing the path type isn't getting rid of the path furniture, maybe it shouldn't magically repair the path furniture, either?

As the code stands now, it looks like if you changed the path type underneath broken path furniture, it would repair the path furniture for free. Not sure if that's the desired behavior or not, since that would normally cost money to fix.

Reply:
Good catch! I think the repair flag should be set to false when resetting the Addition to 0.